### ISSUE TITLE
[BUG] `extend()` remove the '?' from optional type after upgrading 3.20.3 -> 3.20.4/3.20.5

### ISSUE THREAD

**Issue by panmenghan** [colinhacks/zod#2016]
[Playground](https://www.typescriptlang.org/play?ts=4.9.5#code/JYWwDg9gTgLgBAbzgLzgXzgMyhEcDkyEAJvgFBkDGEAdgM7wCudAplHALwoB0EARgCsWlGAAoEwYgC4eNRiD5tRASgA0cGgEMQLGcm4MowGgHMVvMDGC1NAGxVplZOHG4sAHjBY1i4miwB3AGkWAE89bj4ICFsWTRpzCEtrLXtlRwoYULAWOABVVnYufWNMNgAeLJyITDhmNgA+CgB6ZrgANzY6FJkAZm4AJgAGbgAWOGg4fuHuAFYyKtyCtgB9SkYoKG8YFc6obtpORGcXSRk5BTYAbjgTjW1dOENjEzgAHzqfFkxjFmIbuCtODlAC0BAA-Pg4FsQBBOsQ4HxQgQPF4fCpyC4NIEQuFEdFYvF3p9iN9fv9AW1QRCoTC4X9Ecj8KjvL5lOQ0C02nsDjQ+oMRr0FtkloUVmAtu1rMxdl0UkcEHczhp5IooDc7lodOCZM9TMTGF8fv4KZqcWEdfiYnEaAajeSrmQ0EA)

```typescript
import { z } from 'zod'

const user = z.object({id: z.number(), name: z.string().optional()})
  .extend({newKey: z.boolean().optional()})

type User = z.infer<typeof user>

// version: 3.20.4 or 3.20.5
type User_current_version = {
   id: number; 
   name: string | undefined;  // <- '?' removed by 'extend()'
   newKey: boolean | undefined; // <- '?' removed by 'extend()'
}

// version: 3.20.3
type User_previous_version = {
   id: number; 
   name?: string | undefined; 
   newKey?: boolean | undefined;
}
```

**Comment by nghieptiki** [comment]
Same issue with `.merge()`

**Comment by simylein** [comment]
Is this a regression or intentional behaviour?

**Comment by colinhacks** [comment]
Apologies, fixed in 3.20.6.
